scipy/stats/tests/test_stats.py

```diff
@@ -9372,7 +9399,7 @@ def test_empty(self, xp):
 
     def test_dtype(self, xp):
         max = xp.finfo(xp.float32).max
-        x_np = np.asarray([max, max], dtype=np.float32)
+        x_np = np.asarray([max, max/2], dtype=np.float32)
```

Comment:
What is this change for?

Reply:
I mentioned during the `_xp_var` PR that something I wanted to do was make it emit a warning (like `moment`) when precision loss due to catastrophic cancellation is detected. By factoring out `_demean` from `_moment` in this PR, it was easy to use it in `_xp_var`, so this case now emits a warning. I could keep the test the same except for ignoring or testing the warning, but I think changing the test is fine because I don't think it still tests the same thing.